components/aws/sagemaker/common/sagemaker_component.py

```diff
@@ -0,0 +1,569 @@
+"""Base class for all SageMaker components."""
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import re
+import signal
+import string
+import logging
+import json
+from types import FunctionType
+import yaml
+import random
+from pathlib import Path
+from time import sleep, strftime, gmtime
+from abc import abstractmethod
+from typing import Any, Type, Dict, List, NamedTuple, Optional
+
+from .sagemaker_component_spec import SageMakerComponentSpec
+from .boto3_manager import Boto3Manager
+from .common_inputs import (
+    SageMakerComponentBaseOutputs,
+    SageMakerComponentCommonInputs,
+    SpotInstanceInputs,
+)
+
+# This handler is called whenever the @ComponentMetadata is applied.
+# It allows the command line compiler to detect every component spec class.
+_component_decorator_handler: Optional[FunctionType] = None
+
+
+def ComponentMetadata(name: str, description: str, spec: object):
+    """Decorator for SageMaker components.
+
+    Used to define necessary metadata attributes about the component which will
+    be used for logging output and for the component specification file.
+
+    Usage:
+    ```python
+    @ComponentMetadata(
+        name="SageMaker - Component Name",
+        description="A cool new component we made!",
+        spec=MyComponentSpec
+    )
+    """
+
+    def _component_metadata(cls):
+        cls.COMPONENT_NAME = name
+        cls.COMPONENT_DESCRIPTION = description
+        cls.COMPONENT_SPEC = spec
+
+        # Add handler for compiler
+        if _component_decorator_handler:
+            return _component_decorator_handler(cls) or cls
+        return cls
+
+    return _component_metadata
+
+
+class SageMakerJobStatus(NamedTuple):
+    """Generic representation of a job status."""
+
+    is_completed: bool
+    raw_status: str
+    has_error: bool = False
+    error_message: Optional[str] = None
+
+
+class SageMakerComponent(object):
```

Comment:
All classes in python3 are new-style and extend object by default. Why do you need to specify it here ?